module3/homework/advanced_calculator/advancedCalculator.cpp

```diff
@@ -0,0 +1,72 @@
+#include "advancedCalculator.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <map>
+#include <regex>
+
+const std::map<std::string, std::function<double(double, double)>> commands{
+    {"+", std::plus<double>()},
+    {"-", std::minus<double>()},
+    {"*", std::multiplies<double>()},
+    {"/", std::divides<double>()},
+    {"%", std::modulus<int>()},
+    {"^", [](double base, double exponent) { return pow(base, exponent); }},
+    {"$", [](double num, double root) { return pow(num, 1.0 / root); }}
+};
```

Comment:
My suggestion:

You can put factorial there :~}. Let's try sth like that.

{"!", [](const double value, const double defaultValue) { return (value < 0) ? defaultValue : tgamma(value + defaultValue); }};

Value of variable defaultValue is "1". You can define this variable as constexpr for example. Second option is to define this variable as const variable inside factorial case (more info about it - > below)

Reply:
Or just `{'!', [](double lhs, double rhs) { return factorial(lhs); }},` and inside factorial() call gamma